core/src/hir/elision.rs

```diff
@@ -187,11 +168,11 @@ impl ElisionSource {
 ///
 /// This contains data for generating fresh elided lifetimes, looking up named
 /// lifetimes, and caching lifetimes of `Self`.
-struct BaseLifetimeLowerer<'ast> {
-    elided_node_gen: ImplicitLifetimeGenerator,
+pub(super) struct BaseLifetimeLowerer<'ast> {
     lifetime_env: &'ast ast::LifetimeEnv,
     self_lifetimes: Option<TypeLifetimes>,
-    nodes: SmallVec<[LifetimeNode; 2]>,
+    nodes: SmallVec<[Lifetime; 4]>,
+    num_lifetimes: usize,
```

Comment:
nit: comment explaining why this isn't just nodes.len, and explaining how the elided stuff works